File: Pertemuan 10/Pertemuan 11/membuatarraynim.cpp
#include <iostream>
#include <vector>
#include <string>
#include <numeric>
#include <algorithm>
#include <cmath>
#include <iomanip> 

using namespace std;

void hitungNIMArray() {
    string nimStr;
    cout << "Masukkan NIM: ";
    cin >> nimStr;

    vector<int> digit;
    for (char c : nimStr) {
     
        digit.push_back(c - '0'); 
    }

    
    cout << "Digit Array: [";
    for (size_t i = 0; i < digit.size(); ++i) {
        cout << digit[i] << (i < digit.size() - 1 ? ", " : "");
    }
    cout << "]" << endl;

    
    int total = accumulate(digit.begin(), digit.end(), 0);


    int maks = *max_element(digit.begin(), digit.end());
    int minim = *min_element(digit.begin(), digit.end());

    
    double rata = (double)total / digit.size();


    vector<int> rev = digit; 
    reverse(rev.begin(), rev.end()); 

   
    cout << "Total: " << total << endl;
    cout << "Maksimum: " << maks << endl;
    cout << "Minimum: " << minim << endl;
    cout << fixed << setprecision(2) << "Rata-rata: " << rata << endl; 

  
    cout << "Reverse Array: [";
    for (size_t i = 0; i < rev.size(); ++i) {
        cout << rev[i] << (i < rev.size() - 1 ? ", " : "");
    }
    cout << "]" << endl;
}

int main() {
    hitungNIMArray();
    return 0;
}

File: Pertemuan 10/Pertemuan 11/membuatarraynim.py
# SOAL - Membuat Array Dari Digit NIM

# 1. Input NIM
nim = input("Masukkan NIM: ")

# 2. Ubah string menjadi array digit (integer)
# Menggunakan List Comprehension untuk efisiensi
digit = [int(c) for c in nim]

# --- Tugas Khusus: Hitung Menggunakan Loop ---

# #Total
total = 0
for d in digit:
    total += d

# #Maks
# Inisialisasi dengan elemen pertama
maks = digit[0]
for d in digit:
    if d > maks:
        maks = d

# #Min
# Inisialisasi dengan elemen pertama
minim = digit[0]
for d in digit:
    if d < minim:
        minim = d

# #Rata
# Pastikan array tidak kosong
if len(digit) > 0:
    rata = total / len(digit)
else:
    rata = 0

# #Reverse array
rev = []
# Loop dari indeks terakhir (len(digit)-1) hingga 0, dengan step -1
for i in range(len(digit)-1, -1, -1):
    # Kesalahan minor pada gambar: di gambar Anda tertulis rev.append(digit[i]), 
    # namun di potongan lain (3fee5bf7-7651-4731-a045-e3e8f6c36e81.jpg) ada typo digit[1]. 
    # Saya perbaiki menjadi yang benar: digit[i]
    rev.append(digit[i]) 
    
# --- Output Hasil ---
print("Digit        :", digit)
print("Total        :", total)
print("Maksimum     :", maks)
print("Minimum      :", minim)
print("Rata-rata    :", rata)
print("Reverse array:", rev)